orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/manifest/ManifestEvaluator.java

```diff
@@ -0,0 +1,198 @@
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+import retrofit.client.Response;
+
+@Component
+@Slf4j
+@RequiredArgsConstructor
+public class ManifestEvaluator implements CloudProviderAware {
+  private static final ThreadLocal<Yaml> yamlParser =
+      ThreadLocal.withInitial(() -> new Yaml(new SafeConstructor()));
+
+  private final ArtifactResolver artifactResolver;
+  private final OortService oort;
+  private final ObjectMapper objectMapper;
+  private final ContextParameterProcessor contextParameterProcessor;
+  private final KatoService kato;
+
+  private final RetrySupport retrySupport = new RetrySupport();
+
+  @RequiredArgsConstructor
+  @Getter
+  public static class Result {
+    private final List<Map<Object, Object>> manifests;
+    private final List<Artifact> requiredArtifacts;
+    private final List<Artifact> optionalArtifacts;
+  }
+
+  public Result evaluate(Stage stage, ManifestContext context) {
+    Iterable<Object> rawManifests = Collections.emptyList();
+    if (ManifestContext.Source.Artifact.equals(context.getSource())) {
+      Artifact manifestArtifact =
+          artifactResolver.getBoundArtifactForStage(
+              stage, context.getManifestArtifactId(), context.getManifestArtifact());
+
+      if (manifestArtifact == null) {
+        throw new IllegalArgumentException("No manifest artifact was specified.");
+      }
+
+      // Once the legacy artifacts feature is removed, all trigger expected artifacts will be
+      // required to define an account up front.
+      if (context.getManifestArtifactAccount() != null) {
+        manifestArtifact.setArtifactAccount(context.getManifestArtifactAccount());
+      }
+
+      if (manifestArtifact.getArtifactAccount() == null) {
+        throw new IllegalArgumentException("No manifest artifact account was specified.");
+      }
+
+      log.info("Using {} as the manifest", manifestArtifact);
+
+      rawManifests =
+          retrySupport.retry(
+              () -> {
+                Response manifestText = oort.fetchArtifact(manifestArtifact);
+                try (InputStream body = manifestText.getBody().in()) {
+                  return yamlParser.get().loadAll(body);
+                } catch (Exception e) {
+                  log.warn("Failure fetching/parsing manifests from {}", manifestArtifact, e);
+                  // forces a retry
+                  throw new IllegalStateException(e);
+                }
+              },
+              10,
+              200,
+              true); // retry 10x, starting at .2s intervals);
+    } else if (context.getRawManifest() != null) {
+      rawManifests = Collections.singletonList(yamlParser.get().loadAll(context.getRawManifest()));
```

Comment:
Manifests that are stored directly on the stage in `manifests` are already in JSON so we shouldn't be deserializing YAML here.

Reply:
it appears the previous implementation also deserializes the directly stored manifest on the stage in order to perform `ExpressionEvaluation` if `skipExpressionEvaluation` was not set to true

Reply:
Where is that happening?  I'm pretty sure this will error as we'd be parsing JSON as YAML, but maybe I'm missing a translation somewhere---in any case, it's important to test this works for both inline manifests and artifacts.

Reply:
I was incorrect. I made the appropriate changes to maintain previous functionality